gufe/tests/conftest.py

```diff
@@ -22,14 +22,21 @@
 
 ## helper functions
 
+class URLFileLike:
+    def __init__(self, url, encoding='utf-8'):
+        self.url = url
+        self.encoding = encoding
+        self.data = None
 
-def load_url_data(url, encoding='utf-8'):
-    if not HAS_INTERNET:
-        pytest.skip("Skipping because internet seems faulty")
+    def __call__(self):
+        if not HAS_INTERNET:  # pragma: no-cover
+            pytest.skip("Skipping because internet seems faulty")
 
-    req = urllib.request.urlopen(url)
-    # we convert to StringIO because req.read() returns bytes, not string
-    return io.StringIO(req.read().decode(encoding))
+        if self.data is None:
+            req = urllib.request.urlopen(self.url)
+            self.data = req.read().decode(self.encoding)
+
+        return io.StringIO(self.data)
```

Comment:
This is fun, but why can't we use a pytest session scoped fixture here?  Otherwise it's an odd pattern that slows down anyone trying to work on these tests

Reply:
Goals:

1. Have reasonable test labels so that if one PDB test fails, we immediately see the name of the file that fails.
2. Don't have the boilerplate of writing a new fixture for every PDB file we get from a URL.
3. Don't require downloading *all* PDB files if you're only testing one of them.
4. Only download URLs once per session.

Is there a way to accomplish all 4 goals with pytest fixtures? I don't immediately see it. A single fixture to create a dict with all URLs breaks 3. A fixture per URL breaks 2.

Existing code accomplishes 1-3, this change adds 4.